src/gettext.h

```diff
@@ -77,3 +78,20 @@ inline std::wstring fwgettext(const char *src, Args&&... args)
 	delete[] str;
 	return std::wstring(buf);
 }
+
+template <typename ...Args>
+inline std::string fmtgettext(const char *format, std::size_t buf_size, Args&&... args)
+{
+	format = gettext(format);
+	if (buf_size <= 0) {
+		buf_size = porting::mt_snprintf(nullptr, 0, format, std::forward<Args>(args)...) + 1;
+		if (buf_size <= 0) {
+			throw std::runtime_error("gettext format error: " + std::string(format));
+		}
+	}
+	auto size = static_cast<size_t>(buf_size);
```

Comment:
pointless, `buf_size` is already of type size_t

Reply:
For the `mt_snprintf` return int type. But I should make it clearer.